commit fb2240799f3c
Author: Jean-Yves Avenard <jyavenard@mozilla.com>
Date:   Tue Oct 10 16:14:24 2017 +0200

    Bug 1407243 - Don't immediately fire canplaythrough for infinite/live streams. r=jwwang, a=ritu
    
    MozReview-Commit-ID: 2LQFa8PHvO5
    
    --HG--
    extra : source : 6f2fa0f0d78132f5411f648f7c787c2aeeffbd69
---
 dom/media/mediasource/MediaSourceDecoder.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git dom/media/mediasource/MediaSourceDecoder.cpp dom/media/mediasource/MediaSourceDecoder.cpp
index d85f7ec51226..54c7d79e9bdd 100644
--- dom/media/mediasource/MediaSourceDecoder.cpp
+++ dom/media/mediasource/MediaSourceDecoder.cpp
@@ -310,11 +310,7 @@ MediaSourceDecoder::CanPlayThroughImpl()
   }
   TimeUnit duration = TimeUnit::FromSeconds(mMediaSource->Duration());
   auto currentPosition = CurrentPosition();
-  if (duration.IsInfinite()) {
-    // We can't make an informed decision and just assume that it's a live
-    // stream
-    return true;
-  } else if (duration <= currentPosition) {
+  if (duration <= currentPosition) {
     return true;
   }
   // If we have data up to the mediasource's duration or 10s ahead, we can
